File: samples/dotnet-isolated/RedisPubSubTrigger/ResolvedChannelPubSubTrigger.cs
﻿using Google.Protobuf;
using Microsoft.Extensions.Logging;

namespace Microsoft.Azure.Functions.Worker.Extensions.Redis.Samples.RedisPubSubTrigger
{
    internal class ResolvedChannelPubSubTrigger
    {
        private readonly ILogger<ResolvedChannelPubSubTrigger> logger;

        public ResolvedChannelPubSubTrigger(ILogger<ResolvedChannelPubSubTrigger> logger)
        {
            this.logger = logger;
        }

        [Function(nameof(ResolvedChannelPubSubTrigger))]
        public void Run(
            [RedisPubSubTrigger(Common.connectionString, "%pubsubChannel%")] Common.ChannelMessage channelMessage)
        {
            logger.LogInformation($"Message receieved from channel '{channelMessage.Channel}': {channelMessage.Message}");
        }
    }
}

File: samples/java/src/main/java/com/function/RedisPubSubTrigger/ResolvedChannelPubSubTrigger.java
package com.function.RedisPubSubTrigger;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.redis.annotation.*;

public class ResolvedChannelPubSubTrigger {
    @FunctionName("ResolvedChannelPubSubTrigger")
    public void run(
            @RedisPubSubTrigger(
                name = "req",
                connection = "redisConnectionString",
                channel = "%pubsubChannel%",
                pattern = false)
                String message,
            final ExecutionContext context) {
            context.getLogger().info(message);
    }
}
